app/src/ereferrals/java/org/eyeseetea/malariacare/data/remote/PullOrganisationCredentials.java

```diff
@@ -0,0 +1,127 @@
+package org.eyeseetea.malariacare.data.remote;
+
+import android.content.Context;
+import android.net.ConnectivityManager;
+import android.net.NetworkInfo;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import org.eyeseetea.malariacare.R;
+import org.eyeseetea.malariacare.data.IDataSourceCallback;
+import org.eyeseetea.malariacare.data.database.model.Program;
+import org.eyeseetea.malariacare.data.database.utils.PreferencesState;
+import org.eyeseetea.malariacare.domain.entity.Credentials;
+import org.eyeseetea.malariacare.domain.exception.InvalidCredentialsException;
+import org.eyeseetea.malariacare.domain.exception.NetworkException;
+import org.eyeseetea.malariacare.domain.exception.PullConversionException;
+import org.eyeseetea.malariacare.network.ServerAPIController;
+import org.hisp.dhis.client.sdk.models.attribute.AttributeValue;
+import org.hisp.dhis.client.sdk.models.organisationunit.OrganisationUnit;
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.io.IOException;
+
+public class PullOrganisationCredentials {
+
+    public void pullOrganisationCredentials(Credentials credentials,
+            final IDataSourceCallback<Credentials> callback) {
+        boolean isNetworkAvailable = isNetworkAvailable();
+
+        if (!isNetworkAvailable) {
+            callback.onError(new NetworkException());
+        } else {
+            try {
+            JSONObject jsonObject = ServerAPIController.getOrganisationUnitsByCode(
+                    credentials.getUsername());
+                if (jsonObject == null) {
+                    callback.onError(new InvalidCredentialsException());
+                } else {
+                    Credentials OUCredentials = parseOrganisationUnitToCredentials(jsonObject);
```

Comment:
the decision of convert OrgUnit data to credentials is a Domain decision not data decision. This decision should be realized second step in LoginUseCase after login with hardcoded credentials